internal/storage/image.go

```diff
@@ -398,6 +408,145 @@ func (svc *imageService) ImageStatus(systemContext *types.SystemContext, nameOrI
 	return &result, nil
 }
 
+// nolint:staticcheck
+func (svc *imageService) IsRunningImageAllowed(ctx context.Context, systemContext *types.SystemContext, userSpecifiedImage string) error {
+	userSpecifiedImageParsed, err := reference.ParseNormalizedNamed(userSpecifiedImage)
+	if err != nil {
+		return fmt.Errorf("parsing user specified image: %w", err)
+	}
+	userSpecifiedImageParsed = reference.TagNameOnly(userSpecifiedImageParsed)
+
+	policy, err := signature.DefaultPolicy(systemContext)
+	if err != nil {
+		return fmt.Errorf("get default policy: %w", err)
+	}
+
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return fmt.Errorf("create policy context: %w", err)
+	}
```

Comment:
`policyContext.Destroy` needs to be called eventually.

Reply:
Added